as2_behaviors/as2_behaviors_trajectory_generation/generate_polynomial_trajectory_behavior/src/generate_polynomial_trajectory_behavior.cpp

```diff
@@ -89,10 +89,43 @@ DynamicPolynomialTrajectoryGenerator::DynamicPolynomialTrajectoryGenerator(
   RCLCPP_INFO(this->get_logger(), "Sampling with n = %d and dt = %f", sampling_n_, sampling_dt_);
 
   /** Debug publishers **/
-  ref_point_pub = this->create_publisher<visualization_msgs::msg::Marker>(
-    REF_TRAJ_TOPIC, 1);
+  std::string path_topic;
+  std::string reference_setpoint;
+  std::string reference_end_waypoint;
+  std::string reference_waypoints;
+  this->declare_parameter<std::string>("debug.path_topic", "");
+  this->declare_parameter<std::string>("debug.reference_setpoint", "");
+  this->declare_parameter<std::string>("debug.reference_end_waypoint", "");
+  this->declare_parameter<std::string>("debug.reference_waypoints", "");
+  this->get_parameter("debug.path_topic", path_topic);
```

Comment:
@perezsaura-david points out that is not needed to do both declare and after do the get_parameter, instead you could do:
```
path_topic = this->declare_parameter<std::string>("debug.path_topic", "");
```
reducing the number of lines and help readability